src/dask_histogram/boost.py

```diff
@@ -242,12 +190,73 @@ def dask_name(self) -> str:
 
     @property
     def dask(self) -> HighLevelGraph:
+        if self._dask_name == "__not_yet_calculated__" and self._dask is None:
+            self._build_taskgraph()
         if self._dask is None:
             raise RuntimeError(
                 "The dask graph should never be None when it's requested."
             )
         return self._dask
 
+    def _build_taskgraph(self):
+        data_list = []
+        weights = []
+        samples = []
+
+        dask_data = tuple(
+            datum
+            for datum in (
+                self._staged[0]["args"] + tuple(self._staged[0]["kwargs"].values())
+            )
+            if is_dask_collection(datum)
+        )
+
+        if is_dask_awkward_like(dask_data[0]):
+
+            for afill in self._staged:
+                data_list.append(afill["args"])
+                weights.append(afill["kwargs"]["weight"])
+                samples.append(afill["kwargs"]["sample"])
+
+            if all(weight is None for weight in weights):
+                weights = None
+
+            if not all(sample is None for sample in samples):
+                samples = None
+
+            split_every = self._split_every
+            if split_every is None:
+                split_every = dask.config.get("histogram.aggregation.split-every", 8)
+
+            fills = _partitioned_histogram_multifill(
+                data_list, self._histref, weights, samples
+            )
+
+            output_hist = _reduction(fills, split_every)
+        else:
+
+            first_fill = self._staged.pop()
```

Comment:
This is equivalent to the previous behaviour, right? So in some future case, we may still want to know what to do with, for example, dask.array fills.

Reply:
this has been entirely changed now - all input variants use the multifill strategy now.